Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift

```diff
@@ -39,16 +41,22 @@ extension SyntaxStringInterpolation {
 
 public protocol HasTrailingCodeBlock {
   var body: CodeBlockSyntax { get set }
+
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws
 }
 
-public extension HasTrailingCodeBlock where Self: SyntaxExpressibleByStringInterpolation {
-  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) {
-    self = "\(signature) {}"
+public extension HasTrailingCodeBlock where Self: StmtSyntaxProtocol {
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws {
+    let stmt = StmtSyntax("\(signature) {}")
+    guard let castedStmt = stmt.as(Self.self) else {
+      throw SyntaxStringInterpolationError.producedInvalidNodeType(expectedType: IfStmtSyntax.self, actualNode: stmt)
+    }
+    self = castedStmt
     self.body = CodeBlockSyntax(statements: bodyBuilder())
   }
 }
 
-extension CatchClauseSyntax: HasTrailingCodeBlock {}
+//extension CatchClauseSyntax: HasTrailingCodeBlock {}
```

Comment:
Should this be added or removed?

Reply:
Oh, I forgot about this. It should be added. Thanks